First draft spatial Decomposition updates

AlexBrad1eyCT opened the pull request:
Fixes #230 

First Draft of documentation changes to address new form of spatial decomposition

AlexBrad1ey committed:
First draft spatial Decomposition updates

larswik commented on IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md:
```diff
@@ -1,9 +1,9 @@
+The project spatial structure may be made up of a selection of different spatial structure elements with the most generic and simplest form from high to low level as follows: _IfcProject_, _IfcSite_, _IfcFacility_ (or any of its subtypes), _IfcFacilityPart_ (or _IfcBuildingStorey in the case of buildings), and _IfcSpace_ with _IfcSite_, _FacilityPart_ and _IfcSpace_ being optional levels. Therefo ...
 
-In addition a more general hierarchical tree of spatial elements can be created by using _IfcSpatialZone_, from high to low: _IfcProject_, _IfcSite_, and _IfcSpatialZone_ with _IfcSite_ being an optional level.
+Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with preefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevent and agreed typing identifier defined in _IfcFacility.ObjectType_. this allows for nearly full cove ...
```
```suggestion
Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with preefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full coverage of built environment domains and/or scenarios that have yet to be addressed with specific extensions.
```

AlexBrad1eyCT committed:
Update IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md

Co-authored-by: larswik <lars.wikstrom@triona.se>

MatthieuPERIN commented on IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md:
```diff
@@ -1,9 +1,9 @@
+Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with preefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full cove ...
 
-> NOTE&nbsp; The more general hiearchical tree has been introduced as an intermediate solution and stub for further extensions to support infrastructure works.
+In addition to the identified spatial structure elements _IfcSpatialZone_ can be used to provide cross domain or functional zones within a project, these elements can be mounted into the hierarchy in the same manner as other spatial structure elements, but should not aggregate any other spatial structure element types except other _IfcSpatialZone_ instances.
```
Not agree, as far as I know the relation to mount SpatialZone into SpatialStructure is with IfcRelContainedInSpatialStructure and not IfcRelAggragate, that should be stated clearly ?

AlexBrad1eyCT replied:
You are correct @MatthieuPERIN, my mistake i miss read our diagram and have updated this.

MatthieuPERIN reviewed (commented):
Need to clarify the part about IfcSpatialZone and "mounting" it inside Spatial structure.

  
I can provide a remplacement text if needed !

AlexBrad1eyCT committed:
response to comments & update to Spatial Decomposition

updated the Spatial Decomposition CT to compliment the Spatial Composition CT and changed mounting around _IfcSpatialZone_

larswik commented on IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md:
```diff
@@ -1,9 +1,9 @@
+The project spatial structure may be made up of a selection of different spatial structure elements with the most generic and simplest form from high to low level as follows: _IfcProject_, _IfcSite_, _IfcFacility_ (or any of its subtypes), _IfcFacilityPart_ (or _IfcBuildingStorey in the case of buildings), and _IfcSpace_ with _IfcSite_, _FacilityPart_ and _IfcSpace_ being optional levels. Therefo ...
 
-In addition a more general hierarchical tree of spatial elements can be created by using _IfcSpatialZone_, from high to low: _IfcProject_, _IfcSite_, and _IfcSpatialZone_ with _IfcSite_ being an optional level.
+Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with predefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full cov ...
```
```suggestion
Where possible the relevant subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with predefined or user defined type is not available the higher level, generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full coverage of built environment domains and/or scenarios that have yet to be addressed with specific extensions.
```

larswik commented on IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md:
```diff
@@ -1,9 +1,9 @@
+Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with predefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full cov ...
 
-> NOTE&nbsp; The more general hiearchical tree has been introduced as an intermediate solution and stub for further extensions to support infrastructure works.
+In addition to the identified spatial structure elements _IfcSpatialZone_ can be used to provide cross domain or functional zones within a project, these elements are mounted into the hierarchy using the _Spatial Containment_ concept (_IfcRelContainedInSpatialStructure_) and can be aggregated into a functional hierarchy in the same manner as other spatial structure elements with the constraint th ...
```
"mounted" => "included"?

AlexBrad1eyCT replied:
```suggestion
In addition to the identified spatial structure elements _IfcSpatialZone_ can be used to provide cross domain or functional zones within a project, these elements are included into the hierarchy using the _Spatial Containment_ concept (_IfcRelContainedInSpatialStructure_) and can be aggregated into a functional hierarchy in the same manner as other spatial structure elements with the constraint that an _IfcSpatialZone_ can only composed under another _IfcSpatialZone_.
```

larswik commented on IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md:
```diff
@@ -1,9 +1,9 @@
+The project spatial structure may be made up of a selection of different spatial structure elements with the most generic and simplest form from low to high level as follows: _IfcSpace_, _IfcFacilityPart_ (or _IfcBuildingStorey in the case of buildings), _IfcFacility_ (or any of its subtypes), _IfcSite_ and  _IfcProject_ with _IfcSite_, _FacilityPart_ and _IfcSpace_ being optional levels. Therefo ...
 
-In addition a more general hierarchical tree of spatial elements can be created by using _IfcSpatialZone_, from low to high: _IfcSpatialZone_, _IfcSite_, and _IfcProject_, with _IfcSite_ being an optional level.
+Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with predefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full cov ...
```
```suggestion
Where possible the relevant subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with predefined or user defined type is not available the higher level, generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full coverage of built environment domains and/or scenarios that have yet to be addressed with specific extensions.
```

larswik commented on IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md:
```diff
@@ -1,9 +1,9 @@
+Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with predefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full cov ...
 
-> NOTE&nbsp; The more general hiearchical tree has been introduced as an intermediate solution and stub for further extensions to support infrastructure works.
+In addition to the identified spatial structure elements _IfcSpatialZone_ can be used to provide cross domain or functional zones within a project, these elements are mounted into the hierarchy using the _Spatial Containment_ concept (_IfcRelContainedInSpatialStructure_) and can be aggregated into a functional hierarchy in the same manner as other spatial structure elements with the constraint th ...
```
"mounted" => "included"??

AlexBrad1eyCT replied:
```suggestion
In addition to the identified spatial structure elements _IfcSpatialZone_ can be used to provide cross domain or functional zones within a project, these elements are included into the hierarchy using the _Spatial Containment_ concept (_IfcRelContainedInSpatialStructure_) and can be aggregated into a functional hierarchy in the same manner as other spatial structure elements with the constraint that an _IfcSpatialZone_ can only composed under another _IfcSpatialZone_.
```

AlexBrad1eyCT committed:
Update IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md

Co-authored-by: larswik <lars.wikstrom@triona.se>

AlexBrad1eyCT committed:
Update IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md

Co-authored-by: larswik <lars.wikstrom@triona.se>

AlexBrad1eyCT committed:
Update IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md

Co-authored-by: larswik <lars.wikstrom@triona.se>

AlexBrad1eyCT committed:
Update IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md

Co-authored-by: larswik <lars.wikstrom@triona.se>

AlexBrad1eyCT commented:
Changes have been submitted a review reminder for @MatthieuPERIN @SergejMuhic @larswik so we can merge this CT In

MatthieuPERIN commented on IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md:
```diff
@@ -1,9 +1,9 @@
+> NOTE&nbsp; The link between the project and the highest level spatial element is provided by this concept through _IfcRelAggregates_ and not through declaration using _IfcRelDeclares_. This is a known anomaly intruduced to maintain compatibility with earlier versions of this standard.
 
-The order of spatial structure elements being included in the concept for builing projects are from low to high level: _IfcSpace_, _IfcBuildingStorey_, _IfcBuilding_, _IfcSite_ and _IfcProject_ with _IfcSite_, _IfcBuildingStorey_ and _IfcSpace_ being optional levels. Therefore an spatial structure element can only be part of an element at the same or higher level.
+The project spatial structure may be made up of a selection of different spatial structure elements with the most generic and simplest form from low to high level as follows: _IfcSpace_, _IfcFacilityPart_ (or _IfcBuildingStorey in the case of buildings), _IfcFacility_ (or any of its subtypes), _IfcSite_ and  _IfcProject_ with _IfcSite_, _FacilityPart_ and _IfcSpace_ being optional levels. Therefo ...
```
Typos:
higjh -> high

MatthieuPERIN reviewed (changes requested):
One typo, the rest is perfect !

AlexBrad1eyCT committed:
Update IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md

MatthieuPERIN reviewed (approved):
Perfect !
